addons/website/static/lib/multirange/multirange_custom.js

```diff
@@ -295,11 +295,19 @@ class Multirange {
     }
 
     formatNumber(number) {
-        number = String(number).split('.');
-        if (number[1] && number[1].length === 1) {
-            number[1] += '0';
+        console.log("Format number", number);
+        const locale = document.querySelector("html").getAttribute("lang");
+        let formatedNumber;
+        // We use a try catch in case the lang is not supported by the browser.
+        try {
+            formatedNumber = number.toLocaleString(locale, {
+                minimumFractionDigits: 2,
+                maximumFractionDigits: 2,
+            });
```

Comment:
I put this in a `try catch` because I'm a bit worried about doing it without it. If the first parameter given to `toLocaleString` contains special characters other than `-`, (for example `fr_BE`) an error is thrown... so I find it safer to do it this way and avoid having an error displayed to all visitors on "/shop" if the language isn't correctly formatted. Maybe we could remove the try catch in master... I'm waiting for a reviewer's opinion.

Note that I didn't find a way to check if the locale is supported before using it or to get a list of all suported locale.. it would be better but I think there is now way..

Reply:
Note that the python side formats locales the “POSIX way”─i.e. with an underscore separating the different subtags─while the JavaScript APIs conform to the [Unicode BCP 47 locale identifiers](https://unicode.org/reports/tr35/#BCP_47_Conformance), that is, they use hyphens as separators. When dealing with locales from the server, I usually replace underscores with hyphens, otherwise an error will indeed be thrown.

If you want to check for the availability of a locale, you could use [`Intl.NumberFormat.supportedLocalesOf`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/supportedLocalesOf). This probably conveys the intent better than the `try catch`.

Reply:
Yes maybe I can replace underscores by hyphens but I cannot use `Intl.NumberFormat.supportedLocalesOf`. I'm still not sure what is the best todo because underscore is not the only char that throw an error unfortunately..
![image](https://github.com/odoo/odoo/assets/78849981/bced4471-44a3-4053-8959-83cb532a7239)

Reply:
Just replace the underscores beforehand; is there any case where we put some other strange characters in the lang attribute?

Which made me realise: [this attribute is already supposed to contain BCP 47 language tags](https://www.w3.org/International/articles/language-tags/); we are already doing something wrong with those to end up with underscores there. I'll keep this in a corner of my mind for a possible fix in master 😅

Reply:
AFAIK with the standard Odoo code we will never have underscore so in master I'm in favor of simply removing the try catch. On the other hand, in stable, I don't know what kind of messed-up custo the customer might have on his DB. If, by some misfortune, a customer has a special character in the language of his /shop page, I'd rather avoid this commit trowing a traceback to all those visitors on /shop. Maybe I'm being over-cautious here, I don't know... 
If you will r+, I let you decide :)

Reply:
Saw the mails, just passing by:

> we are already doing something wrong with those to end up with underscores there. I'll keep this in a corner of my mind for a possible fix in master 

It's supposed to be impossible, you can ignore the case where someone would have made a very weird custo modifying that.

The `lang` attribute of the `html` tag is [using the lang code](https://github.com/odoo/odoo/blame/6ddd981734686a2ac8f5279b8f464738dec1f4ea/odoo/addons/base/models/ir_qweb.py#L887) (which [can't be modified](https://github.com/odoo/odoo/blame/6ddd981734686a2ac8f5279b8f464738dec1f4ea/odoo/addons/base/models/res_lang.py#L299)) and we make sure to [replace](https://github.com/odoo/odoo/blame/6ddd981734686a2ac8f5279b8f464738dec1f4ea/addons/website/views/website_templates.xml#L57) those `_` by `-` in accordance to [W3C spec](https://www.w3.org/TR/html4/struct/dirlang.html#langcodes)

Reply:
Ok I will ignore this case. Thanks :)